### PR TITLE
Integration Quality Scale indicator

### PR DESCRIPTION
**Pull request by wendevlin**
## Proposed change
- show integration quality scale on integration detail page
  - info link leads to [developers.home-assistant.io](https://developers.home-assistant.io/docs/core/integration-quality-scale)

platinum

gold

silver

bronze

Shimmer effect:
[shimmer.webm](https://github.com/user-attachments/assets/d1b63431-564e-4675-9b1f-35c9436c2d96)

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/integrations/ha-config-integration-page.ts [hunk 1]
```diff
@@ -338,6 +341,36 @@ class HaConfigIntegrationPage extends SubscribeMixin(LitElement) {
...
+                        path=${this._manifest.quality_scale === "platinum"
+                          ? mdiTrophy
+                          : mdiMedal}
+                      ></ha-svg-icon>
+                      ${this.hass.localize(
+                        `ui.panel.config.integrations.config_entry.${this._manifest.quality_scale}_quality`
+                      )}
+                      <a
+                        slot="action"
+                        href=${`https://developers.home-assistant.io/docs/core/integration-quality-scale/#-${this._manifest.quality_scale}`}
```
Comment (frenck) [inline src/panels/config/integrations/ha-config-integration-page.ts:363]: We need to link to the end-user version from our UI:

<https://www.home-assistant.io/docs/quality_scale/>

(that page will be updated for the final release).

../Frenck

File: src/panels/config/integrations/ha-config-integration-page.ts [hunk 2]
```diff
@@ -338,6 +341,36 @@ class HaConfigIntegrationPage extends SubscribeMixin(LitElement) {
                 ${this._manifest?.version != null
                   ? html`<div class="version">${this._manifest.version}</div>`
                   : nothing}
+                ${this._manifest?.quality_scale &&
+                this._manifest?.quality_scale !== "internal"
+                  ? html`<ha-alert
+                      class=${classMap({
```
Comment (bramkragten) [inline src/panels/config/integrations/ha-config-integration-page.ts:347]: No need to use a classMap here, just use a template

File: src/panels/config/integrations/ha-config-integration-page.ts (hunk 1 above)
Comment (bramkragten) [inline src/panels/config/integrations/ha-config-integration-page.ts:363]: Use the `documentationUrl` helper

**Review by MindFreeze — DISMISSED** [review]

File: src/panels/config/integrations/ha-config-integration-page.ts [hunk 3]
```diff
@@ -338,6 +341,36 @@ class HaConfigIntegrationPage extends SubscribeMixin(LitElement) {
                 ${this._manifest?.version != null
                   ? html`<div class="version">${this._manifest.version}</div>`
                   : nothing}
+                ${this._manifest?.quality_scale &&
+                this._manifest?.quality_scale !== "internal"
```
[the remark this answers is not in the document]
Comment (wendevlin) [reply src/panels/config/integrations/ha-config-integration-page.ts:345]: :grin:  good catch

[the remark this answers is not in the document]
Comment (frenck) [reply src/panels/config/integrations/ha-config-integration-page.ts:345]: We do not need to check that, will handle that case in core. As task is open for that.
The IQS also describes this case, that is will be automatically assigned to any custom integration.

File: src/panels/config/integrations/ha-config-integration-page.ts (hunk 1 above)
Comment (wendevlin) [reply src/panels/config/integrations/ha-config-integration-page.ts]: I'll use it, makes the code clearer

File: src/panels/config/integrations/ha-config-integration-page.ts [hunk 4]
```diff
@@ -338,41 +340,72 @@ class HaConfigIntegrationPage extends SubscribeMixin(LitElement) {
...
-                  : ""}
+                      )}
+                    </div>`
+                  : nothing}
                 ${this._manifest?.iot_class?.startsWith("cloud_")
-                  ? html`<ha-alert
-                      ><ha-svg-icon slot="icon" path=${mdiCloud}></ha-svg-icon
-                      >${this.hass.localize(
+                  ? html`<div class="integration-info">
+                      <ha-svg-icon .path=${mdiWeb}></ha-svg-icon>
```
Comment (bramkragten) [inline src/panels/config/integrations/ha-config-integration-page.ts:390]: If we update the icon here, we should also update it on the integration page and the new integration picker

Comment (wendevlin) [reply src/panels/config/integrations/ha-config-integration-page.ts:390]: That icon is already there. Just in integration detail it was a cloud.

File: src/panels/config/integrations/ha-config-integration-page.ts [hunk 5]
```diff
@@ -338,41 +340,72 @@ class HaConfigIntegrationPage extends SubscribeMixin(LitElement) {
                 ${this._manifest?.version != null
                   ? html`<div class="version">${this._manifest.version}</div>`
                   : nothing}
+                ${this._manifest?.quality_scale &&
+                this._manifest?.quality_scale !== "internal"
+                  ? html`
+                      <div class="quality-scale integration-info">
```
Comment (bramkragten) [inline src/panels/config/integrations/ha-config-integration-page.ts:346]: You should use `ha-list-item` so we are sure it aligns with the items underneath it.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**